plugins/promela/hu.bme.mit.gamma.promela.verification/src/hu/bme/mit/gamma/promela/verification/PromelaVerifier.xtend

```diff
@@ -192,11 +196,11 @@ class PromelaQueryAdapter {
 	}
 	
 	def adaptQuery(String query) {
-		if (query.startsWith("E")) {
+		if (query.startsWith(EF)) {
 			invert = true
-			return "!(" + query.substring(E.length) + " )"
+			return G + "(!(" + query.substring(EF.length) + ") )"
```

Comment:
I think the original solution was correct, you just had to handle the `<>` operator as an `F` operator.